interfaces/builtin/posix_mq.go

```diff
@@ -0,0 +1,291 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const posixMQSummary = `allows access to POSIX message queues`
+
+// This interface is super-privileged
+const posixMQBaseDeclarationSlots = `
+  posix-mq:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+// Paths can only be specified by the slot and a slot needs to exist for the
+// plug to connect to, so the plug does not need to be marked super-privileged
+const posixMQBaseDeclarationPlugs = `
+  posix-mq:
+    allow-installation: true
+    allow-connection:
+      slot-attributes:
+        posix-mq: $PLUG(posix-mq)
+    allow-auto-connection:
+      slot-publisher-id:
+        - $PLUG_PUBLISHER_ID
+      slot-attributes:
+        posix-mq: $PLUG(posix-mq)
```

Comment:
afaict there is no code atm checking that posix-mq is set to a string, or defaulting it to the slot/plug name as we do for content and shared-memory

Reply:
If I understand you correctly this is already in the getPath function:

https://github.com/snapcore/snapd/pull/11590/files/4d14c3028cb30aece59556dcf696332a13d3e906#diff-81c542b03cd89ab9e295c71416ac4d48c89b02af12f513d203fe313e7d89aa2fR151

Reply:
no,   the attribue referred here is called "posix-mq", it's labeling the slot to match up with consuming plugs (I imagine). It's not "path". I suggest looking how shared_memory.go deals with the labeling "shared-memory" attribute vs a "path" attribute.